components/consentRequestForm/index.jsx

```diff
@@ -207,10 +209,11 @@ export default function ConsentRequestForm({ agentDetails, agentWebId }) {
     setSelectedDate(null);
     setDatepickerOpen(false);
   };
-
   useEffect(() => {
     if (expirationDate) {
       setSelectedDate(new Date(expirationDate));
+    } else {
+      setDateForever();
     }
   }, [expirationDate]);
```

Comment:
I don't think you actually need this, as `expirationDate` is set when `selectedDate` is Initialised (L102-104), and as `consentRequest` comes from context, if the consent request changes, this whole component will re-render, so `expirationDate` will never change.

You could maybe shift the date parsing logic down to the DateInput and have it just return a string for the date, either "Forever" or a [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) formatted date string (I assume that's what the VC spec would call for).

That way you don't need to marshal the date value here and you don't have multiple setState's for the expiration.

Also, `selectedDate` isn't actually read anywhere in the `handleSubmit` function, I assume it's meant to be.